docs/rfcs/XeGPU.md

```diff
@@ -285,7 +245,201 @@ In case that certain Xe GPU target does not support atomic operation for a certa
 Attribute `scope` describes the scope of fence. "workgroup" means that the scope is within each work group. "gpu" means the scope is across work groups within the gpu.
 Attribute `Memory_kind` describes the memory kind. "global" means the global memory, "shared" means the shared local memory.
 
-`nbarrier` and `fence` operations lower to uniform instructions, so there is no need to specify the sg_map or VC mode.
+`nbarrier` and `fence` operations lower to uniform instructions, so there is no need to specify the `sg_map`.
+
+## XeGPU Attributes to support Work Item Level semantics  
+
+Attribute `xegpu.sg_map` describes the mapping between work item (WI) WI threads and the 2D tensor specified by the tensor descriptor. To distribute the XeGPU operation to WI threads, the tensor_desc must be specified with the `sg_map` attribute at the tensor description creation time.
+
+Within the sg_map, `wi_layout` specifies the layout of WI threads, describing the mapping of WI threads to the tensor. wi_layout[0] x wi_layout[1] must be equal to the total number of WI threads within a subgroup. `wi_data` specifies the data elements assigned to each WI thread for a single distribution. In the example below, the subgroup has 16 WI threads in wi_layout=[1, 16], each accessing 1 element as specified by wi_data=[1,1]. 
+
+```mlir
+  #sg_map_a = xegpu.sg_map<wi_layout = [1, 16], wi_data = [1, 1]>
+  %tdesc_a = XeGPU.create_nd_tdesc %mem_addr, %offsets:2, %base_shape:2,%base_stride:2
+		: uint64, index, index, index, index, index, index
+     	into tensor_desc<8x16xbf16, #sg_map_a>
+```
+
+wi_data_size refers to the data size mapped to indiviudal WI therad, and sg_map_size to the collective size by all the WI threads as specified by sg_map. sg_map_size represents the minimun size of 2D tensor to be distributed to WI threads. tensor_desc_size refers to the size of the tensor sepcified by tensor_desc. 
+In the example above, wi_data_size is 1, sg_map_size is 16, tensor_desc_size is 128. 
+```mlir
+        wi_data_size = wi_data[0] x wi_data[1]
+	SIMD_LANE == wi_layout[0] x wi_layout[1] 
+ 	sg_maps_size = SIMD_LANE * wi_data_size 
+  	tensor_desc_size = tensor_desc[0] x tensor_desc[1]
+```
+
+size distribution rule can be represented as following: tensor_desc_size[0] must be divisible by wi_layout[0] x wi_data[0], tensor_desc_size[1] must be divisible by wi_layout[1] x wi_data[1]. The 2D subtensor is evenly distributed to WI threads, so each WI thread gets a 2D data fragments. 
```

Comment:
Not sure I got  why tensor_desc_size becomes 2d - isn't it defined above as a scalar?

Reply:
When XeGPU.load_nd(2d) load a tensor_dec to vector, the load is a 2D block load which processes  2D tile data.

Reply:
Ahh. Now I understand your comments. Changed.